Stop excluding functions with callable parameter synopsis

Duroth opened the pull request:
The generator checks for overloaded methods (functions with multiple signatures) and excludes these from the auto-generated functions.

However, this check errors on functions that include a callable in their parameters; in many cases, the callable's signature gets a synopsis of its own on the docs, which means the generator erroneously interprets it as an overloaded function.

This PR aims to fix this edge case by only pulling the synopsis from the "Description" section of the docpage, and excluding the "Parameters" part. This should mean that any synopses pertaining to the function's parameters, rather than the function itself, are now skipped and no longer trip up the generator.

dbrekelmans commented on generated/image.php:
```diff
@@ -1772,33 +1772,13 @@ function imagegrabscreen()
- * @throws ImageException
- *
- */
-function imagegrabwindow(int $handle, bool $client_area = false)
```
This should not have been removed: https://www.php.net/manual/en/function.imagegrabwindow

dbrekelmans commented on generated/simplexml.php:
```diff
@@ -1,87 +0,0 @@
- * @throws SimplexmlException
- *
- */
-function simplexml_import_dom(\DOMNode $node, string $class_name = SimpleXMLElement::class): \SimpleXMLElement
```
This file should not have been removed. If the root cause is not related to the issue you're fixing, you could manually add it back (the build will fail, but I can still merge it and look at it in a different PR)

dbrekelmans commented on generated/spl.php:
```diff
@@ -4,69 +4,6 @@
- * @throws SplException
- *
- */
-function class_implements($class, bool $autoload = true): array
```
This file should not have been removed

dbrekelmans commented on generated/url.php:
```diff
@@ -28,35 +28,6 @@ function base64_decode(string $string, bool $strict = false): string
- * @throws UrlException
- *
- */
-function get_headers(string $url, bool $associative = false, $context = null): array
```
This file should not have been removed

dbrekelmans commented on generator/src/DocPage.php:
```diff
@@ -163,9 +163,17 @@ public function getMethodSynopsis(): array
+
+        // Only evaluate the synopsis inside the `<refsect1 role="description">...</refsect1>` section of the doc page.
+        // Other synopses might occur in the `<refsect1 role="parameters">...</refsect1>` section, but these describe
+        // handlers, callbacks, and other callable-type arguments, not the function itself.
```
Nice fix! This adds a few functions that were not generated before :+1:

dbrekelmans commented on rector-migrate.php:
```diff
@@ -372,7 +369,6 @@
             'imagegd2' => 'Safe\imagegd2',
             'imagegif' => 'Safe\imagegif',
             'imagegrabscreen' => 'Safe\imagegrabscreen',
-            'imagegrabwindow' => 'Safe\imagegrabwindow',
```
Keep in mind, if you revert unjust removals to also revert those function removals in this file

dbrekelmans reviewed (changes requested):
Thanks @Duroth! There are some removals by the generator which might or might not be related to your change. If they are not related, you can revert the removals. The build will fail, but I can still merge it and look for the root cause through a different PR.

Duroth commented on generated/dom.php:
```diff
@@ -11,7 +11,7 @@
  *
  * @param \SimpleXMLElement $node The SimpleXMLElement node.
- * @return \DOMElement The DOMElement node added.
+ * @return \DOMElement|false The DOMElement node added.
```
You can blame phpstan for this. ;)

Duroth commented on generated/array.php:
```diff
@@ -47,7 +47,7 @@ function array_combine(array $keys, array $values): array
  *
  */
-function array_flip(array $array): array
+function array_flip($array): array
```
Probably due to phpstan changing the parameters; PHP-Safe borrows parameter info from phpstan, and array_flip is one of the only functions I can find that is mapped with typed arrays:

```
'array_flip' => ['array', 'input'=>'array<int|string>'],
```

Just about all other functions use a generic `array` type hint.

Duroth commented:
- I've reverted the `.gitignore` changes;
- I've also reverted the generated code. While this may go against contributing guidelines, the (already) failing build is a good reminder of what functions still need to be looked at. I feel there's plenty more to be done before we can confidently re-generate a new baseline source.

dbrekelmans commented:
Thansk @Duroth!
